qwerty_ast/src/ast.rs

```diff
@@ -377,6 +445,15 @@ impl Func {
     }
 }
 
+impl std::fmt::Display for Func {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Func::Qpu(def) => write!(f, "@qpu {def}"),
+            Func::Classical(def) => write!(f, "@classical {def}"),
```

Comment:
Wow great idea, seriously
```suggestion
            Func::Qpu(def) => write!(f, "@qpu\n{def}"),
            Func::Classical(def) => write!(f, "@classical\n{def}"),
```